packages/core/core/src/services/entity-service/index.ts

```diff
@@ -146,275 +77,84 @@ const createDefaultImplementation = ({
-    }
-
-    entity = await this.wrapResult(entity, { uid, action: 'create' });
-
-    const { ENTRY_CREATE } = ALLOWED_WEBHOOK_EVENTS;
-    await this.emitEvent(uid, ENTRY_CREATE, entity);
-
-    return entity;
+    return this.wrapResult(entity, { uid, action: 'create' });
   },
 
   async update(uid, entityId, opts) {
     const wrappedParams = await this.wrapParams<
-      EntityService.Params.Pick<typeof uid, 'data:partial' | 'files' | 'fields' | 'populate'>
+      EntityService.Params.Pick<typeof uid, 'data:partial' | 'fields' | 'populate'>
     >(opts, {
       uid,
       action: 'update',
     });
-    const { data, files } = wrappedParams;
-
-    const model = strapi.getModel(uid);
-
     const entityToUpdate = await db.query(uid).findOne({ where: { id: entityId } });
 
     if (!entityToUpdate) {
-      return null;
+      return this.wrapResult(null, { uid, action: 'update' });
     }
 
-    const isDraft = contentTypesUtils.isDraft(entityToUpdate, model);
+    const shouldPublish = !contentTypesUtils.isDraft(entityToUpdate, strapi.getModel(uid));
 
-    const validData = await entityValidator.validateEntityUpdate(
-      model,
-      data,
-      {
-        isDraft,
-      },
-      entityToUpdate
-    );
-    // @ts-expect-error - fix
-    const query = transformParamsToQuery(uid, pickSelectionParams(wrappedParams));
-
-    // TODO: wrap in transaction
-    const componentData = await updateComponents(uid, entityToUpdate, validData);
-    const entityData = updatePipeline(
-      Object.assign(omitComponentData(model, validData), componentData),
-      { contentType: model }
-    );
-
-    let entity = await db.query(uid).update({
-      ...query,
-      where: { id: entityId },
-      data: entityData,
+    const entity = strapi.documents!(uid).update(entityToUpdate.documentId, {
+      ...(wrappedParams as any),
+      status: shouldPublish ? 'published' : 'draft',
     });
 
-    // TODO: upload the files then set the links in the entity like with compo to avoid making too many queries
-    if (files && Object.keys(files).length > 0) {
-      await this.uploadFiles(uid, Object.assign(entityData, entity), files);
-      entity = await this.findOne(uid, entity.id, wrappedParams);
-    }
-
-    entity = await this.wrapResult(entity, { uid, action: 'update' });
-
-    const { ENTRY_UPDATE } = ALLOWED_WEBHOOK_EVENTS;
-    await this.emitEvent(uid, ENTRY_UPDATE, entity);
-
-    return entity;
+    return this.wrapResult(entity, { uid, action: 'update' });
   },
 
   async delete(uid, entityId, opts) {
     const wrappedParams = await this.wrapParams(opts, { uid, action: 'delete' });
 
-    // select / populate
-    // @ts-expect-error - fix
-    const query = transformParamsToQuery(uid, pickSelectionParams(wrappedParams));
-
-    let entityToDelete = await db.query(uid).findOne({
-      ...query,
-      where: { id: entityId },
-    });
+    const entityToDelete = await db.query(uid).findOne({ where: { id: entityId } });
```

Comment:
minor thing, should we support deleting components? Or things that are not content types (media for example, which will be in the future but not now).

Reply:
I meant directly using the component uid to delete it (same as for create and update).
It is just a random thought to provide more v4 compatibilty